crates/stages/stages/src/stages/prune.rs

```diff
@@ -0,0 +1,158 @@
+use reth_db_api::database::Database;
+use reth_provider::DatabaseProviderRW;
+use reth_prune::{PruneModes, PrunerBuilder};
+use reth_stages_api::{
+    ExecInput, ExecOutput, Stage, StageCheckpoint, StageError, StageId, UnwindInput, UnwindOutput,
+};
+use tracing::info;
+
+/// The prune stage that runs the pruner with the provided prune modes.
+///
+/// There are two main reasons to have this stage when running a full node:
+/// - Sender Recovery stage inserts a lot of data into the database that's only needed for the
+///   Execution stage. Pruner will clean up the unneeded recovered senders.
+/// - Pruning during the live sync can take a significant amount of time, especially history
+///   segments. If we can prune as much data as possible in one go before starting the live sync, we
+///   should do it.
+///
+/// `commit_threshold` is the maximum number of entries to prune before committing
+/// progress to the database.
+#[derive(Debug)]
+pub struct PruneStage {
+    prune_modes: PruneModes,
+    commit_threshold: usize,
+}
+
+impl PruneStage {
+    /// Crate new prune stage with the given prune modes and commit threshold.
+    pub const fn new(prune_modes: PruneModes, commit_threshold: usize) -> Self {
+        Self { prune_modes, commit_threshold }
+    }
+}
+
+impl<DB: Database> Stage<DB> for PruneStage {
+    fn id(&self) -> StageId {
+        StageId::Prune
+    }
+
+    fn execute(
+        &mut self,
+        provider: &DatabaseProviderRW<DB>,
+        input: ExecInput,
+    ) -> Result<ExecOutput, StageError> {
+        let mut pruner = PrunerBuilder::default()
+            .segments(self.prune_modes.clone())
+            .prune_max_blocks_per_run(1)
```

Comment:
terminology is weird here, `prune_max_blocks_per_run` is for calculating the deletion limit based on how many blocks were between previous and current pruner runs. but the actual limit is calculated as `delete_limit_per_block * min(current_tip - previous_tip, prune_max_blocks_per_run)`

here, we do `1` because we want just `delete_limit_per_block * 1`

Reply:
this needs to be improved